autoload/maktaba/extension.vim

```diff
@@ -0,0 +1,159 @@
+""
+" @dict ExtensionRegistry
+" A registry for the extensions used a single plugin.  Extensions are
+" dictionaries with a plugin-specific interpretation
+"
+" Plugins should use |Plugin.GetExtensionRegistry| to gain access to their own
+" extension registry, and @function(maktaba#extension#GetRegistry) to gain
+" access to an extension registry for another plugin.
+
+
+" Extension registry objects, keyed by plugin name.
+if !exists('s:registries')
+  let s:registries = {}
+endif
+
+
+""
+" @private
+" Returns an extension registry for the given plugin name, creating it if
+" needed.
+" @throws WrongType if {plugin} is not a string.
+function! maktaba#extension#GetInternalRegistry(plugin) abort
+  call maktaba#ensure#IsString(a:plugin)
+
+  if has_key(s:registries, a:plugin)
+    return s:registries[a:plugin]
+  endif
+
+  " TODO: should this contain the plugin's name?
+  " TODO: is there a better way to construct a do-nothing _validator?
```

Comment:
I don't think there is a better way right now. I've wanted one before just for tinkering around at runtime with code that expects a function, so it would be a reasonable feature request. But in plugins it's usually less weird to just leave it undefined and guard the callsite with `has_key(l:registry, '_validator')`. Or this is fine if you want to leave it.